source/postcard-schema/src/max_size.rs

```diff
@@ -0,0 +1,124 @@
+        DataModelType::Char => Some(5), // I think? 1 len + up to 4 bytes
+        DataModelType::String => None,
+        DataModelType::ByteArray => None,
+        DataModelType::Option(nt) => max_size_nt(nt),
+        DataModelType::Unit => Some(0),
+        DataModelType::UnitStruct => Some(0),
+        DataModelType::NewtypeStruct(nt) => max_size_nt(nt),
+        DataModelType::Seq(_) => None,
+        DataModelType::Tuple(nts) | DataModelType::TupleStruct(nts) => {
+            let mut i = 0;
+            let mut ct = 0;
+            while i < nts.len() {
+                let Some(sz) = max_size_nt(nts[i]) else {
+                    return None;
+                };
+                ct += sz;
+                i += 1;
+            }
+            Some(ct)
+        }
+        DataModelType::Map { .. } => None,
+        DataModelType::Struct(nvals) => {
+            let mut i = 0;
+            let mut ct = 0;
+            while i < nvals.len() {
+                let Some(sz) = max_size_dmt(nvals[i].ty.ty) else {
+                    return None;
+                };
+                ct += sz;
+                i += 1;
+            }
+            Some(ct)
+        }
+        DataModelType::Enum(nvars) => {
+            let mut i = 0;
+            let mut max = 0;
+            while i < nvars.len() {
+                let sz = match nvars[i].ty {
+                    DataModelVariant::UnitVariant => 0,
+                    DataModelVariant::NewtypeVariant(nt) => {
+                        let Some(sz) = max_size_nt(nt) else {
+                            return None;
+                        };
+                        sz
+                    }
+                    DataModelVariant::TupleVariant(nts) => {
+                        let mut j = 0;
+                        let mut ct = 0;
+                        while j < nts.len() {
+                            let Some(sz) = max_size_nt(nts[j]) else {
+                                return None;
+                            };
+                            ct += sz;
+                            j += 1;
+                        }
+                        ct
+                    }
+                    DataModelVariant::StructVariant(nvars) => {
+                        let mut j = 0;
+                        let mut ct = 0;
+                        while j < nvars.len() {
+                            let Some(sz) = max_size_dmt(nvars[j].ty.ty) else {
+                                return None;
+                            };
+                            ct += sz;
+                            j += 1;
+                        }
+                        ct
+                    }
+                };
+
+                if sz > max {
+                    max = sz;
+                }
+
+                i += 1;
+            }
+            let disc_size = if nvars.is_empty() {
+                1
+            } else {
+                // CEIL(variants / 7)
+                (nvars.len() + 6) / 7
```

Comment:
This isn't right, it should be `2^7` not `7`, we want to check how many variants fit in how many varint bytes.